domain/src/main/kotlin/no/nav/su/se/bakover/domain/søknadsbehandling/KanOppdaterePeriodeBosituasjonVilkår.kt

```diff
@@ -0,0 +1,337 @@
+package no.nav.su.se.bakover.domain.søknadsbehandling
+
+import arrow.core.Either
+import arrow.core.getOrElse
+import arrow.core.left
+import arrow.core.right
+import no.nav.su.se.bakover.common.extensions.toNonEmptyList
+import no.nav.su.se.bakover.common.ident.NavIdentBruker
+import no.nav.su.se.bakover.common.tid.Tidspunkt
+import no.nav.su.se.bakover.common.tid.periode.erSammenhengendeSortertOgUtenDuplikater
+import no.nav.su.se.bakover.domain.grunnlag.EksterneGrunnlagSkatt
+import no.nav.su.se.bakover.domain.grunnlag.Grunnlag
+import no.nav.su.se.bakover.domain.grunnlag.GrunnlagsdataOgVilkårsvurderinger
+import no.nav.su.se.bakover.domain.grunnlag.avslåPgaOpplysningsplikt
+import no.nav.su.se.bakover.domain.sak.Sakstype
+import no.nav.su.se.bakover.domain.søknadsbehandling.stønadsperiode.Aldersvurdering
+import no.nav.su.se.bakover.domain.søknadsbehandling.vilkår.KunneIkkeLeggeTilVilkår
+import no.nav.su.se.bakover.domain.søknadsbehandling.vilkår.VilkårsfeilVedSøknadsbehandling
+import no.nav.su.se.bakover.domain.vilkår.FamiliegjenforeningVilkår
+import no.nav.su.se.bakover.domain.vilkår.FastOppholdINorgeVilkår
+import no.nav.su.se.bakover.domain.vilkår.FlyktningVilkår
+import no.nav.su.se.bakover.domain.vilkår.FormuegrenserFactory
+import no.nav.su.se.bakover.domain.vilkår.InstitusjonsoppholdVilkår
+import no.nav.su.se.bakover.domain.vilkår.KunneIkkeLageOpplysningspliktVilkår
+import no.nav.su.se.bakover.domain.vilkår.LovligOppholdVilkår
+import no.nav.su.se.bakover.domain.vilkår.OpplysningspliktVilkår
+import no.nav.su.se.bakover.domain.vilkår.PensjonsVilkår
+import no.nav.su.se.bakover.domain.vilkår.PersonligOppmøteVilkår
+import no.nav.su.se.bakover.domain.vilkår.UføreVilkår
+import no.nav.su.se.bakover.domain.vilkår.UtenlandsoppholdVilkår
+import no.nav.su.se.bakover.domain.vilkår.VurdertVilkår
+import no.nav.su.se.bakover.domain.vilkår.formue.LeggTilFormuevilkårRequest
+import java.time.Clock
+
+/**
+ * TODO jah: Mulig å splitte denne inn i 3 interfaces KanOppdatereStønadsperiode, KanOppdatereBosituasjon og KanOppdatereVilkår
+ */
+sealed interface KanOppdaterePeriodeBosituasjonVilkår : Søknadsbehandling, KanOppdaterePeriodeGrunnlagVilkår {
```

Comment:
Oppsettet her så litt rart ut. Men interfacet er KanOppdaterePeriodeBosituasjonVilkår, mens det finnes funksjoner for en rekke andre ulike ting?